components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Engine/LegacyPublisherConfigServicesTests.cs

```diff
@@ -267,5 +270,139 @@ public class LegacyPublisherConfigServicesTests {
         }
+
+        [Fact]
+        public async Task PublishNodesStressTest() {
+            var legacyCliModelProviderMock = new Mock<ILegacyCliModelProvider>();
+            var agentConfigProviderMock = new Mock<IAgentConfigProvider>();
+            var identityMock = new Mock<IIdentity>();
+            var newtonSoftJsonSerializer = new NewtonSoftJsonSerializer();
+            var jobSerializer = new PublisherJobSerializer(newtonSoftJsonSerializer);
+            var publishedNodesJobConverter = new PublishedNodesJobConverter(TraceLogger.Create(), newtonSoftJsonSerializer);
+
+            string tempPublishedNodesFile = Path.GetTempFileName();
+            using (var fileStream = new FileStream(tempPublishedNodesFile, FileMode.Open, FileAccess.Write)) {
+                fileStream.Write(Encoding.UTF8.GetBytes("[]"));
+            }
+
+            var legacyCliModel = new LegacyCliModel {
+                PublishedNodesFile = tempPublishedNodesFile,
+                PublishedNodesSchemaFile = "Storage/publishednodesschema.json"
+            };
+            legacyCliModelProviderMock.Setup(p => p.LegacyCliModel).Returns(legacyCliModel);
+            agentConfigProviderMock.Setup(p => p.Config).Returns(new AgentConfigModel());
+
+            var orchestrator = new LegacyJobOrchestrator(
+                publishedNodesJobConverter,
+                legacyCliModelProviderMock.Object,
+                agentConfigProviderMock.Object,
+                jobSerializer,
+                TraceLogger.Create(),
+                identityMock.Object
+            );
+
+            var numberOfEndpoints = 100;
+            var numberOfNodes = 1000;
+
+            var payload = new List<PublishedNodesEntryModel>();
+            for (int endpointIndex = 0; endpointIndex < numberOfEndpoints; ++endpointIndex) {
+                var model = new PublishedNodesEntryModel {
+                    EndpointUrl = new Uri($"opc.tcp://server{endpointIndex}:49580"),
+                };
+
+                model.OpcNodes = new List<OpcNodeModel>();
+                for (var nodeIndex = 0; nodeIndex < numberOfNodes; ++nodeIndex) {
+                    model.OpcNodes.Add(new OpcNodeModel {
+                        Id = $"ns=2;s=Node-Server-{nodeIndex}",
+                    });
+                }
+
+                payload.Add(model);
+            }
+
+            // Publish all nodes.
+            foreach (var request in payload) {
+                var publishNodesResult = await orchestrator.PublishNodesAsync(request).ConfigureAwait(false);
+                publishNodesResult.First()
+                    .Should()
+                    .Be("Succeeded");
+            }
+
+            async Task CheckEndpointsAndNodes(
+                int expectedNumberOfEndpoints,
+                int expectedNumberOfNodes
+            ) {
+                var tasks = new List<Task<JobProcessingInstructionModel>>();
+                for (var i = 0; i < expectedNumberOfEndpoints; i++) {
```

Comment:
you can go expectedNumberOfEndpoints+1 to make sure no extra job was created.

Reply:
Good point. Done.